internal/status/routeconditions.go

```diff
@@ -124,6 +124,29 @@ func (r *RouteParentStatusUpdate) AddCondition(conditionType gatewayapi_v1beta1.
 	return cond
 }
 
+// ConditionExists check the condition is exists or not
+func (r *RouteParentStatusUpdate) ConditionExists(conditionType gatewayapi_v1beta1.RouteConditionType) bool {
+	var rps *gatewayapi_v1beta1.RouteParentStatus
+
+	for _, v := range r.RouteParentStatuses {
+		if v.ParentRef == r.parentRef {
+			rps = v
+			break
+		}
+	}
+	if rps == nil {
+		return false
+	}
+
+	for _, c := range rps.Conditions {
+		if c.Type == string(conditionType) {
+			return true
+		}
+	}
+
+	return false
```

Comment:
Can simplify this using the existing method to:

```suggestion
	for _, c := range r.ConditionsForParentRef(r.parentRef) {
		if c.Type == string(conditionType) {
			return true
		}
	}

	return false
```